examples/gsm8k/gsm8k.py

```diff
@@ -0,0 +1,350 @@
+    folder_name = f"{extra_info}_{timestamp}"
+    results_folder = os.path.join(results_dir, folder_name)
+    os.makedirs(results_folder)
+    config = {
+        "data": selected_data,
+        "methods": [method.__name__ for method in methods],
+        "lm": lm_name,
+        "budget": budget,
+    }
+    with open(os.path.join(results_folder, "config.json"), "w") as f:
+        json.dump(config, f)
+    
+    logging.basicConfig(
+        filename=os.path.join(results_folder, "log.log"),
+        filemode="w",
+        format="%(name)s - %(levelname)s - %(message)s",
+        level=logging.DEBUG,
+    )
+    for method in methods:
+        # create a results directory for the method
+        os.makedirs(os.path.join(results_folder, method.__name__))
+    
+    for data in selected_data:
+        logging.info(f"Running data {data['id']: }{data['question']}: {data['answer']}")
+        if budget <= 0.0:
+            logging.error(
+                f"Budget has been depleted, stopping. Data {data[0]} has not been run."
+            )
+            break
+        for method in methods:
+            logging.info(f"Running method {method.__name__}")
+            logging.info(f"Budget left: {budget}")
+            if budget <= 0.0:
+                logging.error(
+                    f"Budget has been depleted, stopping. Method {method.__name__} has not been run."
+                )
+                break
+            lm = language_models.ChatGPT(
+                os.path.join(
+                    os.path.dirname(__file__),
+                    "../../graph_of_thoughts/language_models/config.json",
+                ),
+                model_name=lm_name,
+                cache=True,
+            )
+            operations_graph = method()
+            executor = controller.Controller(
+                lm,
+                operations_graph,
+                GSM8KPrompter(),
+                GSM8KParser(),
+                {
+                    "original": data["question"],
+                    "ground_truth": utils.strip_int_result(data["answer"]),
+                    "current": "",
+                    "phase": 0,
+                    "method": method.__name__,
+                },
+            )
+            try:
+                executor.run()
+            except Exception as e:
+                logging.error(f"Exception: {e}")
+            path = os.path.join(
+                results_folder,
+                method.__name__,
+                f"{data['id']}.json",
+            )
+            executor.output_graph(path)
+            budget -= lm.cost
+
+    return orig_budget - budget
+
+
+if __name__ == "__main__":
+    budget = 30
+    samples = [item for item in range(30)]
+    approaches = [io, cot, cotsc]
+
+    logging.basicConfig(level=logging.INFO)
+
+    spent = run(samples, approaches, budget, "chatgpt")
+
+    logging.info(f"Spent {spent} out of {budget} budget.")
```

Comment:
newline at the end of file
```suggestion
    logging.info(f"Spent {spent} out of {budget} budget.")
    
```